artiq/coredevice/comm_analyzer.py

```diff
@@ -776,3 +776,9 @@ def decoded_dump_to_target(manager, devices, dump, uniform_interval):
             if isinstance(message, OutputMessage):
                 slack.set_value_double(
                     (message.timestamp - message.rtio_counter)*ref_period)
+
+    if not stopped_messages:
+        logger.warning("StoppedMessage missing")
+    else:
+        end_time = get_message_time(stopped_messages[-1])
```

Comment:
Why would they be of the same time?

Reply:
From my observation, they can have different times.

Reply:
StoppedMessage is generated on analyzer disarm (``enable_write(0)``). That is generated when master analyzer queries the satellites, one by one. They will have differing timestamps; how different will also depend on how many events each satellite before has.

Reply:
Disarm first, transfer, then repeat with the next satellite. There's no attempt at synchronized disarming, since at the stage when the user pulls the analyzer logs they're not in a rush anymore.

It is just the AUX processing indeed, but I would be careful about setting a stiff limit. Well, it would be a warning anyway; it doesn't matter that much.

Reply:
I also don't think it's worth attention from users, so probably debug message is enough

Reply:
Nah not too bad, just need an extra aux message, but that would break back compat. Just not sure what that would bring us; the software still needs to handle extra stopmessages.